extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxInputStream.java

```diff
@@ -270,6 +270,14 @@ protected ByteBuf readBlocking() throws IOException {
         @Override
         public void handle(Buffer event) {
             synchronized (request.connection()) {
+                if (event.length() == 0) {
+                    // When using H2, this indicates that we won't receive anymore data.
```

Comment:
Sorry, but H2 is really confusing (I know it would be weird to use the H2 database to handle HTTP but still):
```suggestion
                    // When using HTTP/2, this indicates that we won't receive anymore data.
```

Reply:
Except that H2 is not HTTP/2:

```
HTTP/2 is defined both for HTTP URIs (i.e. without TLS, a configuration which is abbreviated in h2c) and for HTTPS URIs (over TLS using ALPN extension where TLS 1.2 or newer is required, a configuration which is abbreviated in h2).
```

Reply:
Nice. Let's use an acronym that is even shorter and confusing for something more specific :).

Reply:
This thew me off initially, but it's standard unfortunately...

Reply:
it's an acronym from a standard and not the name of a product :-D 
However, I changed it to be more explicit (HTTP/2 h2)

Reply:
and yeah... What a [wonderful world](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjJw4_Vs4L6AhVrgM4BHZzICTMQyCl6BAgWEAM&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DA3yCcXgbKrE&usg=AOvVaw32pmh31_jnQx1V_fckF_lJ)